### PR TITLE
Port details page to react

### PR DESCRIPTION
**Pull request by RoyEJohnson**
This is my magnum opus.
Issue #1783 and also Issue #1293

### CODE REVIEW

File: src/app/pages/details/desktop-view/desktop-view.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                <ContentGroup activeIndex={labels.indexOf(selectedLabel)}>
+                    <DetailsTab polish={polish} model={model} tocState={tocState} />
+                    <InstructorResourceTab model={model} />
+                    <StudentResourceTab model={model} />
+                    {model.videos[0] && <VideoTab videos={model.videos[0]} />}
+                </ContentGroup>
+            </div>
+        </React.Fragment>
+    );
+}
```
Comment (nathanstitt) [inline src/app/pages/details/desktop-view/desktop-view.js:69]: seems a bit weird to me to be hardcoding references to `polish` language.  What happens if/when we want to support multiple languages?

File: src/app/pages/details/phone-view/instructor-resources-pane/instructor-resources-pane.scss [hunk 2]
```diff
@@ -41,7 +41,7 @@
             @include set-font(h4);
         }
 
-        .resource-box {
+        .Login to unlockx {
             width: 20rem;
```
Comment (nathanstitt) [inline src/app/pages/details/phone-view/instructor-resources-pane/instructor-resources-pane.scss:45]: that doesn't seem like a valid selector?  maybe the result of a  bad search-and-replace?

**Review by nathanstitt — APPROVED** [review]
Looks good Roy, the only real thing I spotted was the invalid selector in the css file

File: src/app/components/tab-group/tab-group.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    // Couldn't get this to work as a normal attribute expression;
+    // it would not remove the aria-current attribute, only its value
+    useEffect(() => {
+        if (label === selectedLabel) {
+            ref.current.setAttribute('aria-current', 'page');
+        } else {
+            ref.current.removeAttribute('aria-current');
+        }
+    }, [label, selectedLabel]);
+
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/tab-group/tab-group.jsx:17]: Thanks, it's a good idea, but for whatever reason, `aria-current` still gets left in the attributes. Could be a Preact bug, but I think I'm stuck with this for now.

File: src/app/pages/details/common/authors.js [hunk 4]
```diff
@@ -25,18 +25,22 @@ export default function AuthorsSection({model, polish}) {
     ] : [
         'Senior Contributing Authors', 'Contributing Authors'
     ];
+    const senior = (author) => author.seniorAuthor;
+    const nonsenior = (author) => !author.seniorAuthor;
+    const allSenior = model.authors.filter(senior);
+    const allNonsenior = model.authors.filter(nonsenior);
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/pages/details/common/authors.js:32]: Handy.

File: src/app/pages/details/desktop-view/desktop-view.js (hunk 1 above)
Comment (RoyEJohnson) [reply src/app/pages/details/desktop-view/desktop-view.js:69]: I was assured that would not happen. If it does, we have a lot of thinking to do about how to support multiple languages. This is basically a technical debt with no foreseeable due date.

File: src/app/pages/details/phone-view/instructor-resources-pane/instructor-resources-pane.scss (hunk 2 above)
Comment (RoyEJohnson) [reply src/app/pages/details/phone-view/instructor-resources-pane/instructor-resources-pane.scss:45]: I don't know what happened there. Thanks for catching it!
Also, thanks for your prompt review of this behemoth.

**Pull request merged**